packages/iTwinUI-react/src/core/Select/Select.tsx

```diff
@@ -254,26 +311,61 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
           key: `${option.label}-${index}`,
           isSelected,
           onClick: () => {
-            !option.disabled && onChange?.(option.value);
-            close();
+            !option.disabled &&
+              onChange?.(
+                option.value,
+                Array.isArray(value) && value.includes(option.value)
+                  ? 'Removed'
+                  : 'Added',
+              );
+            !multiple && close();
+          },
+          ref: (el: HTMLElement) => {
+            if (isSelected && !multiple) {
+              el?.scrollIntoView();
+            }
           },
-          ref: (el: HTMLElement) => isSelected && el?.scrollIntoView(),
           role: 'option',
           ...option,
           ...menuItem.props,
         });
       });
     },
-    [itemRenderer, onChange, options, value],
+    [itemRenderer, multiple, onChange, options, value],
   );
 
-  const selectedItem = React.useMemo(() => {
+  const selectedItems = React.useMemo(() => {
     if (value == null) {
       return undefined;
     }
-    return options.find((option) => option.value === value);
+    return Array.isArray(value)
+      ? options.filter((option) => value.some((val) => val === option.value))
+      : options.find((option) => option.value === value);
   }, [options, value]);
 
+  const selectedItemsArray = React.useMemo(() => {
```

Comment:
Maybe rename this to be `selectedItemsElements` or `selectedItemsJsx` or `selectedItemsTags`.

Reply:
renamed to `selectedItemsElements`